torch/_dynamo/pgo.py

```diff
@@ -0,0 +1,106 @@
+# Profile-guided optimization for Dynamo
+
+from __future__ import annotations
+
+import functools
+import logging
+import os
+import os.path
+import pickle
+from typing import Dict, Optional, TYPE_CHECKING
+
+
+log = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from torch._dynamo.symbolic_convert import InstructionTranslator
+    from torch._dynamo.variables.builder import FrameStateSizeEntry
+
+# TODO: an individual cache file per code object may not be optimal from an IO
+# perspective
+
+
+@functools.lru_cache(None)
+def _get_code_object_cache_path(filename: str, firstlineno: int, name: str) -> str:
+    # TODO: fix inductor layering problem
+    from torch._inductor.codecache import sha256_hash
+    from torch._inductor.runtime.runtime_utils import cache_dir
+
+    # TODO: pickle is too heavy a hammer, do not actually need it
+    # TODO: this scheme makes manual inspection of cache entries difficult,
+    # consider adding some breadcrumbs in the name for ease of use
+    r = os.path.join(
+        cache_dir(), "dynamo", sha256_hash(pickle.dumps((filename, firstlineno, name)))
```

Comment:
`pickle.dumps` is slow, I'd suggest `f"{filename},{firstlineno},{name}".encode("utf-8")`